### ISSUE TITLE
IndexedDB Support

### ISSUE THREAD

**Issue by keean** [SafeExamBrowser/seb-mac#9]
IndexedDB works fine in iOS 10 & 11, but it is not visible in iOS9. iOS9 support is important because there are some devices still in service that cannot be upgraded to iOS10+.

**Comment by keean** [comment]
Some further info, IndexedDB works in Chrome for iOS-9, which uses WkWebView, so it is possible to get it working. It would suggest that there is something missing from the setup of the WkWebView, perhaps an option that was later defaulted to 'on' in iOS-10 explaining why it works on iOS-10+ but not on iOS-9.

**Comment by danschlet** [comment]
We aren’t using WKWebView in SEB for iOS currently. It’s still missing some customizing features like modifying requests and custom server authentication which we will need for some SEB features which will come to the iOS version in the next major update 2.2 (Browser Exam and new Config Key, TLS certificate embedding and pinning and maybe more). Interesting to see that Apple updated UIWebView with IndexedDB in iOS 10, obviously they know themselves that some people still have to use UIWebView...

**Comment by keean** [comment]
Hmm, I am less interested in those features, and just need the device lockdown, my best option might be to make my own browser-shell using WkWebView for iOS9 support. Did you get com.apple.developer.edu-assessment-mode entitlement for SEB? If so how long did it take Apple to respond?

**Comment by danschlet** [comment]
Yes, of course we're using com.apple.developer.edu-assessment-mode, that's the only way of locking down non-supervised iOS devices automatically. We were in direct contact with the relevant people at Apple so we got it quickly after requesting it.

We will support WKWebView in a later version, but I don't expect this support before the next major iOS release (in case Service Workers will require WKWebView, then we might prioritize that).

**Comment by danschlet** [comment]
Can't you use a WebSQL wrapper for IndexedDB?

**Comment by keean** [comment]
Currently IndexedDBShim+WebSQL crashes when I run a few 'puts' in parallel.

**Comment by keean** [comment]
Another issue with UIWebView is the JavaScript is unaccelerated, so any processing operations are much slower than with the WkWebView. Code that takes a few seconds in WkWebView can take more that a minute in the  UIWebView.  

Maybe for those that don't need certain features you could have a configuration option to use a WkWebView, and then disable the options that won't work with it.

**Comment by danschlet** [comment]
Will be fixed when SEB supports WKWebView.
